tests/partitions_test.py

```diff
@@ -114,18 +114,13 @@ def test_isolate_primary_from_one_backup(network, args):
     #   - If b_1 calls first, it can win and then bring _both_ nodes up-to-date, becoming a _stable_ primary
     # So we repeat elections until b_1 is primary
 
-    old_primary = p
-    for _ in range(10):
-        new_primary, new_view = network.wait_for_new_primary(
-            old_primary, nodes=backups, timeout_multiplier=6
-        )
-        if new_primary == b_1:
-            break
-        else:
-            old_primary = new_primary
-
+    new_primary = network.wait_for_primary_unanimity(
+        min_view=initial_txid.view, timeout_multiplier=30
```

Comment:
30 seems very large?

Reply:
It's lower than the 10*6 we potentially had last time. In practice, we often converge much faster, but I have seen ~60s waits in local runs. But if we hit the unlucky timing where 2 very-close elections cancel each other out (why we use 6 for a 'normal' timeout), this still lets us do 5 such elections in a row.